ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/InternalPaywallView.kt

```diff
@@ -37,9 +46,18 @@ internal fun InternalPaywallView(
     }
 }
 
+@Composable
+private fun updateStateIfLocaleChanged(viewModel: PaywallViewModel) {
+    var locale by remember { mutableStateOf(LocaleListCompat.getDefault()) }
+    if (locale != LocaleListCompat.getDefault()) {
+        locale = LocaleListCompat.getDefault()
+        viewModel.refreshState()
+    }
+}
+
 @Composable
 private fun getPaywallViewModel(offering: Offering?, listener: PaywallViewListener?): PaywallViewModel {
     return viewModel<PaywallViewModelImpl>(
-        factory = PaywallViewModelFactory(offering, listener),
+        factory = PaywallViewModelFactory(WeakReference(LocalContext.current.applicationContext), offering, listener),
```

Comment:
Honestly speaking, the application context shouldn't be recreated so it should be ok to hold a strong reference during the app's lifetime... But I thought I would be extra safe and make it weak.